### PR TITLE
Unmapped license errors

### PR DESCRIPTION
**Pull request by mnonnenmacher**
If the declared license of a project or package could not be processed
add an error.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1315)

### CODE REVIEW

File: model/src/main/kotlin/AnalyzerResult.kt [hunk 1]
```diff
@@ -83,6 +83,13 @@ data class AnalyzerResult(
             }
         }
 
+        packages.forEach { curatedPackage ->
```
Comment (sschuberth) [inline model/src/main/kotlin/AnalyzerResult.kt]: Commit message nit: Maybe start the sentence with "Add an error if ...".

File: model/src/main/kotlin/Package.kt [hunk 2]
```diff
@@ -141,6 +141,24 @@ data class Package(
         )
     }
 
+    /**
+     * Check if this package contains any erroneous data.
+     */
+    fun collectErrors(): List<OrtIssue> {
+        val issues = mutableListOf<OrtIssue>()
```
Comment (sschuberth) [inline model/src/main/kotlin/Package.kt]: Should we also use the more compact function expression syntax with `= mutableListOf<OrtIssue>().also { issues ->` here?

File: model/src/main/kotlin/AnalyzerResult.kt (hunk 1 above)
Comment (mnonnenmacher) [reply model/src/main/kotlin/AnalyzerResult.kt]: Done.

File: model/src/main/kotlin/Package.kt (hunk 2 above)
Comment (mnonnenmacher) [reply model/src/main/kotlin/Package.kt]: Done, used `map` to make it even more compact.

**Pull request merged**